src/coreclr/jit/emitxarch.cpp

```diff
@@ -4279,6 +4278,181 @@ bool emitter::IsMovInstruction(instruction ins)
     }
 }
 
+//----------------------------------------------------------------------------------------
+// IsRedundantMov:
+//    Check if the current `mov` instruction is redundant and can be omitted.
+//    A `mov` is redundant in following 3 cases:
+//
+//    1. Move to same register on TARGET_AMD64
+//       (Except 4-byte movement like "mov eax, eax" which zeros out upper bits of eax register)
+//
+//         mov rax, rax
+//
+//    2. Move that is identical to last instruction emitted.
+//
+//         mov rax, rbx  # <-- last instruction
+//         mov rax, rbx  # <-- current instruction can be omitted.
+//
+//    3. Opposite Move as that of last instruction emitted.
+//
+//         mov rax, rbx  # <-- last instruction
+//         mov rbx, rax  # <-- current instruction can be omitted.
+//
+// Arguments:
+//    ins  - The current instruction
+//    fmt  - The current format
+//    size - Operand size of current instruction
+//    dst  - The current destination
+//    src  - The current source
+// canSkip - The move can be skipped as it doesn't represent special semantics
```

Comment:
Consider renaming this to better convey what it means, something like `canIgnoreSideEffects`?

Reply:
I've renamed this in `IsRedundantMov` but not everywhere else (as `/* canSkip */ value` is used for almost every call to `emitIns_Mov`).